apps/gallery/js/startup.js

```diff
@@ -261,23 +257,34 @@
     var placeholder = $('thumbnails-placeholder');
     placeholder.parentElement.replaceChild(thumbnails, placeholder);
 
-    // Handle clicks on the thumbnails
-    thumbnails.addEventListener('click', thumbnailClickHandler);
-
     // When the first page of thumbnails is diplayed, we can emit
     // our 'visually complete' mark for startup time comparison
     Thumbnails.firstpage.then(function() {
       // Tell performance monitors that "above the fold" content is displayed
-      // and is ready to interact with.
       window.performance.mark('visuallyLoaded');
-      window.performance.mark('contentInteractive');
 
       // Remove the attribute to trigger gaia header font-fit logic, defer it
-      // til now to postpone the overhead.
-      var headers = document.querySelectorAll('gaia-header');
-      for (var i = 0; i < headers.length; ++i) {
-        headers[i].removeAttribute('no-font-fit');
-      }
+      // til now to postpone the overhead. Add setTimeout to lazyload
+      // gaia-header.js aynchronously.
+      setTimeout(function() {
+        LazyLoader.load(['shared/elements/gaia-header/dist/gaia-header.js'],
+          function() {
+            var headers = document.querySelectorAll('gaia-header');
+            for (var i = 0; i < headers.length; ++i) {
+              headers[i].removeAttribute('no-font-fit');
+            }
+          });
+      });
+
+      // Clicking on the select button goes to thumbnail select mode
+      $('thumbnails-select-button').onclick =
+        setView.bind(null, LAYOUT_MODE.select);
+
+      // Handle clicks on the thumbnails
+      thumbnails.addEventListener('click', thumbnailClickHandler);
+      // Tell performance monitors that "above the fold" content
+      // is ready to interact with.
+      window.performance.mark('contentInteractive');
```

Comment:
I thought you were going to move this stuff into the lazy load callback.  There doesn't see to be much point in moving it here. According to raptor it makes only about 1ms or less difference in when contentInteractive happens. 

I don't feel strongly about this, but I'd say either leave this code where it was, or do it after the gaia-header code is actually loaded.  In that case, I'd expect to see contentInteractive come 30 to 100ms after visually loaded